// whatwg-url/lib/public-api.js
"use strict";

exports.URL = require("whatwg-url/lib/URL").interface;
exports.URLSearchParams = require("whatwg-url/lib/URLSearchParams").interface;

exports.parseURL = require("whatwg-url/lib/url-state-machine").parseURL;
exports.basicURLParse = require("whatwg-url/lib/url-state-machine").basicURLParse;
exports.serializeURL = require("whatwg-url/lib/url-state-machine").serializeURL;
exports.serializeHost = require("whatwg-url/lib/url-state-machine").serializeHost;
exports.serializeInteger = require("whatwg-url/lib/url-state-machine").serializeInteger;
exports.serializeURLOrigin = require("whatwg-url/lib/url-state-machine").serializeURLOrigin;
exports.setTheUsername = require("whatwg-url/lib/url-state-machine").setTheUsername;
exports.setThePassword = require("whatwg-url/lib/url-state-machine").setThePassword;
exports.cannotHaveAUsernamePasswordPort = require("whatwg-url/lib/url-state-machine").cannotHaveAUsernamePasswordPort;

exports.percentDecode = require("whatwg-url/lib/urlencoded").percentDecode;
