// jsdom/lib/jsdom/living/index.js
"use strict";

exports.DOMException = require("domexception");
exports.NamedNodeMap = require("jsdom/lib/jsdom/living/generated/NamedNodeMap").interface;
exports.Attr = require("jsdom/lib/jsdom/living/generated/Attr").interface;
exports.Node = require("jsdom/lib/jsdom/living/generated/Node").interface;
exports.Element = require("jsdom/lib/jsdom/living/generated/Element").interface;
exports.DocumentFragment = require("jsdom/lib/jsdom/living/generated/DocumentFragment").interface;
exports.Document = exports.HTMLDocument = require("jsdom/lib/jsdom/living/generated/Document").interface;
exports.XMLDocument = require("jsdom/lib/jsdom/living/generated/XMLDocument").interface;
exports.CharacterData = require("jsdom/lib/jsdom/living/generated/CharacterData").interface;
exports.Text = require("jsdom/lib/jsdom/living/generated/Text").interface;
exports.CDATASection = require("jsdom/lib/jsdom/living/generated/CDATASection").interface;
exports.ProcessingInstruction = require("jsdom/lib/jsdom/living/generated/ProcessingInstruction").interface;
exports.Comment = require("jsdom/lib/jsdom/living/generated/Comment").interface;
exports.DocumentType = require("jsdom/lib/jsdom/living/generated/DocumentType").interface;
exports.DOMImplementation = require("jsdom/lib/jsdom/living/generated/DOMImplementation").interface;
exports.NodeList = require("jsdom/lib/jsdom/living/generated/NodeList").interface;
exports.HTMLCollection = require("jsdom/lib/jsdom/living/generated/HTMLCollection").interface;
exports.HTMLOptionsCollection = require("jsdom/lib/jsdom/living/generated/HTMLOptionsCollection").interface;
exports.DOMStringMap = require("jsdom/lib/jsdom/living/generated/DOMStringMap").interface;
exports.DOMTokenList = require("jsdom/lib/jsdom/living/generated/DOMTokenList").interface;

exports.SVGAnimatedString = require("jsdom/lib/jsdom/living/generated/SVGAnimatedString").interface;
exports.SVGNumber = require("jsdom/lib/jsdom/living/generated/SVGNumber").interface;
exports.SVGStringList = require("jsdom/lib/jsdom/living/generated/SVGStringList").interface;

exports.Event = require("jsdom/lib/jsdom/living/generated/Event").interface;
exports.CloseEvent = require("jsdom/lib/jsdom/living/generated/CloseEvent").interface;
exports.CustomEvent = require("jsdom/lib/jsdom/living/generated/CustomEvent").interface;
exports.MessageEvent = require("jsdom/lib/jsdom/living/generated/MessageEvent").interface;
exports.ErrorEvent = require("jsdom/lib/jsdom/living/generated/ErrorEvent").interface;
exports.HashChangeEvent = require("jsdom/lib/jsdom/living/generated/HashChangeEvent").interface;
exports.FocusEvent = require("jsdom/lib/jsdom/living/generated/FocusEvent").interface;
exports.PopStateEvent = require("jsdom/lib/jsdom/living/generated/PopStateEvent").interface;
exports.UIEvent = require("jsdom/lib/jsdom/living/generated/UIEvent").interface;
exports.MouseEvent = require("jsdom/lib/jsdom/living/generated/MouseEvent").interface;
exports.KeyboardEvent = require("jsdom/lib/jsdom/living/generated/KeyboardEvent").interface;
exports.TouchEvent = require("jsdom/lib/jsdom/living/generated/TouchEvent").interface;
exports.ProgressEvent = require("jsdom/lib/jsdom/living/generated/ProgressEvent").interface;
exports.StorageEvent = require("jsdom/lib/jsdom/living/generated/StorageEvent").interface;
exports.CompositionEvent = require("jsdom/lib/jsdom/living/generated/CompositionEvent").interface;
exports.WheelEvent = require("jsdom/lib/jsdom/living/generated/WheelEvent").interface;
exports.EventTarget = require("jsdom/lib/jsdom/living/generated/EventTarget").interface;

exports.BarProp = require("jsdom/lib/jsdom/living/generated/BarProp").interface;
exports.Location = require("jsdom/lib/jsdom/living/generated/Location").interface;
exports.History = require("jsdom/lib/jsdom/living/generated/History").interface;
exports.Screen = require("jsdom/lib/jsdom/living/generated/Screen").interface;
exports.Performance = require("jsdom/lib/jsdom/living/generated/Performance").interface;

exports.Blob = require("jsdom/lib/jsdom/living/generated/Blob").interface;
exports.File = require("jsdom/lib/jsdom/living/generated/File").interface;
exports.FileList = require("jsdom/lib/jsdom/living/generated/FileList").interface;
exports.ValidityState = require("jsdom/lib/jsdom/living/generated/ValidityState").interface;

exports.DOMParser = require("jsdom/lib/jsdom/living/generated/DOMParser").interface;

exports.FormData = require("jsdom/lib/jsdom/living/generated/FormData").interface;
exports.XMLHttpRequestEventTarget = require("jsdom/lib/jsdom/living/generated/XMLHttpRequestEventTarget").interface;
exports.XMLHttpRequestUpload = require("jsdom/lib/jsdom/living/generated/XMLHttpRequestUpload").interface;

exports.NodeIterator = require("jsdom/lib/jsdom/living/generated/NodeIterator").interface;
exports.TreeWalker = require("jsdom/lib/jsdom/living/generated/TreeWalker").interface;

exports.Storage = require("jsdom/lib/jsdom/living/generated/Storage").interface;

require("jsdom/lib/jsdom/living/register-elements")(exports);

// These need to be cleaned up...
require("jsdom/lib/jsdom/level2/style").addToCore(exports);
require("jsdom/lib/jsdom/level3/xpath")(exports);

// This one is OK but needs migration to webidl2js eventually.
require("jsdom/lib/jsdom/living/node-filter")(exports);

exports.URL = require("whatwg-url").URL;
exports.URLSearchParams = require("whatwg-url").URLSearchParams;
